tests/Benchmarks/System.Buffers/Reader_Search.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using System.Buffers.Reader;
+using System.Buffers.Testing;
+using System.Buffers.Text;
+
+namespace System.Buffers.Benchmarks
+{
+    public class Reader_Search
+    {
+        private static byte[] s_array;
+        private static ReadOnlySequence<byte> s_ros;
+        private static ReadOnlySequence<byte> s_rosSplit;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            s_array = new byte[1_000_000];
+            Random r = new Random(42);
+            r.NextBytes(s_array);
```

Comment:
To get deterministic performance numbers, does it make sense to filter out the delimiters we are looking for so that we measure the worst case performance of TryReadUntil?

Currently, it is hard to reason about how TryReadUntil_Sequence compares to TryReadUntilAny_2_Sequence/etc., since its possible that 42 is in segment 1, while 0, 255 are in segment N.

Reply:
> does it make sense to filter out the delimiters

As a separate test, sure.

> it is hard to reason about how TryReadUntil_Sequence 

I would hope that the large set of input data makes this easy to reason about- i.e. it distributes across a pretty wide set of cases. I'm splitting to something like 20,000 segments here.